pkg/services/ngalert/notifier/file_store.go

```diff
@@ -62,6 +63,46 @@ func (fileStore *FileStore) FilepathFor(ctx context.Context, filename string) (s
 	return fileStore.pathFor(filename), err
 }
 
+// GetFullState returns a base64-encoded string containing the Alertmanager's silences and nflog.
+func (fileStore *FileStore) GetFullState(ctx context.Context) (string, error) {
+	all, err := fileStore.kv.GetAll(ctx)
+	if err != nil {
+		return "", err
+	}
+
+	silences, ok := all[fileStore.orgID]["silences"]
```

Comment:
Is this operation safe? 

Looking at the implementation of `GetAll`, I don't think you'll get an error if `GetAll` produces no results which would make this operation panic?

Instead, we should check at the top level map:

`states, ok := all[fileStore.orgID]`

Reply:
Writing to a `nil` map results in a panic, but reading from it doesn't. I'll add a check to return a more accurate error message though, e.g.:

```go
fmt.Errorf("no values for org %d", orgID)
fmt.Errorf("no value for key %q", key)
```